components/shopify_partner/sources/new_app_charges.js

```diff
@@ -0,0 +1,67 @@
+const shopify = require("../shopify_partner.app.js");
+const getAppTransactions = require("../queries/getAppTransactions");
+
+module.exports = {
+  key: "shopify_partner-new-app-charges",
+  name: "New App Charges",
+  type: "source",
+  version: "0.0.1",
+  description:
+    "Emit new events when new app charges made to your partner account.",
+  props: {
+    shopify,
+    db: "$.service.db",
+    createdAtMin: {
+      type: "string",
+      description:
+        "Only include transactions after this specific time (ISO timestamp)",
+      label: "createdAtMin",
+    },
+    createdAtMax: {
+      type: "string",
+      description:
+        "Only include transactions up to this specific time (ISO timestamp)",
+      label: "createdAtMin",
+    },
+    timer: {
+      description: "How often this action should run",
+      type: "$.interface.timer",
+      label: "timer",
+      default: {
+        intervalSeconds: 60 * 60,
+      },
+    },
+  },
+  dedupe: "unique",
```

Comment:
Can be pulled into the common file

Reply:
Sounds like it would work, but interestingly enough the dev command doesn't deploy:

```javascript

  key: "shopify_partner-new-app-installs",
  name: "New App Installs",
  type: "source",
  version: "0.0.1",
  description: "Emit new events when new shops install your app.",
  ...common,
  props: {
    ...common.props,
    db: "$.service.db",
    appId: {
      type: "string",
      optional: false,
      description: "gid://partners/App/<your App ID here>",
      label: "Shopify App ID",
    },
    occurredAtMin: {
      type: "string",
      description:
        "Only include install events after this specific time (ISO timestamp)",
      label: "occurredAtMin",
      optional: true,
    },
    occurredAtMax: {
      type: "string",
      description:
        "Only include install events up to this specific time (ISO timestamp)",
      label: "occurredAtMix",
      optional: true,
    },
  },
```

```bash
error         | component create api call responded with status: 400, body: {"error":"{\"name\":\"UserError\",\"message\":\"unsupported type for prop dedupe: unique\"}"}                                           │
Waiting for file change (or ^C to quit)...
```